quickstarts/microsoft.web/function-app-flex-managed-identities/main.bicep

```diff
@@ -0,0 +1,233 @@
+@description('The maximum scale-out instance count limit for the app.')
+@minValue(40)
+@maxValue(1000)
+param maximumInstanceCount int = 100
+@description('The memory size of instances used by the app.')
+@allowed([2048,4096])
+param instanceMemoryMB int = 2048
+
+@description('A unique token used for resource name generation.')
+@minLength(3)
+param resourceToken string = toLower(uniqueString(subscription().id, location))
+@description('A globally unigue name for your deployed function app.')
+param appName string = 'func-${resourceToken}'
+
+//********************************************
+// Variables
+//********************************************
+
+// Generates a unique container name for deployments.
+var deploymentStorageContainerName = 'app-package-${take(appName, 32)}-${take(resourceToken, 7)}'
+
+// Key access to the storage account is disabled by default 
+var storageAccountAllowSharedKeyAccess = false
+
+// Define the IDs of the roles we need to assign to our managed identities.
+var storageBlobDataOwnerRoleId  = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'
+var storageBlobDataContributorRoleId = 'ba92f5b4-2d11-453d-a403-e96b0029c9fe'
+var storageQueueDataContributorId = '974c5e8b-45b9-4653-ba55-5f855dd0fb88'
+var storageTableDataContributorId = '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3'
+var monitoringMetricsPublisherId = '3913510d-42f4-4e42-8a64-420c390055eb'
+
+//********************************************
+// Azure resources required by your function app.
+//********************************************
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2023-09-01' = {
+  name: 'log-${resourceToken}'
+  location: location
+  properties: any({
+    retentionInDays: 30
+    features: {
+      searchVersion: 1
+    }
+    sku: {
+      name: 'PerGB2018'
+    }
+  })
+}
+
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' = {
+  name: 'appi-${resourceToken}'
+  location: location
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+    WorkspaceResourceId: logAnalytics.id
+  }
+}
+
+resource storage 'Microsoft.Storage/storageAccounts@2023-05-01' = {
+  name: 'st${resourceToken}'
+  location: location
+  kind: 'StorageV2'
+  sku: { name: 'Standard_LRS' }
+  properties: {
+    accessTier: 'Hot'
+    allowBlobPublicAccess: false
+    allowSharedKeyAccess: storageAccountAllowSharedKeyAccess
+    dnsEndpointType: 'Standard'
+    minimumTlsVersion: 'TLS1_2'
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Allow'
+    }
+    publicNetworkAccess: 'Enabled'
+  }
+  resource blobServices 'blobServices' = {
+    name: 'default'
+    properties: {
+      deleteRetentionPolicy: {}
+    }
+    resource deploymentContainer 'containers' = {
+      name: deploymentStorageContainerName
+      properties: {
+        publicAccess: 'None'
+      }
+    }
+  }
+}
+
+resource userStorageDataOwner 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
```

Comment:
Only one user assigned identity is needed, don't need multiple. The user assigned identitiy is associated with the function app. That identity should be given the 4 different storage access roles and also the app insights role, giving the function app access to storage and to send to app insights.